src/apify/storage_clients/_hybrid_apify/_storage_client.py

```diff
@@ -0,0 +1,226 @@
+                `force_cloud` argument when opening storages or automatically when running on the Apify platform.
+            local_storage_client: Client used to communicate with the storage when not running on the Apify
+                platform and not using `force_cloud` argument when opening storages.
+        """
+        self._cloud_storage_client = cloud_storage_client or ApifyStorageClient(request_queue_access='single')
+        self._local_storage_client = local_storage_client or ApifyFileSystemStorageClient()
+
+    def _get_suitable_storage_client(self, *, force_cloud: bool = False) -> StorageClient:
+        if self._is_at_home:
+            return self._cloud_storage_client
+
+        configuration = ApifyConfiguration.get_global_configuration()
+        if force_cloud:
+            if configuration.token is None:
+                raise RuntimeError(
+                    'In order to use the Apify cloud storage from your computer, '
+                    'you need to provide an Apify token using the APIFY_TOKEN environment variable.'
+                )
+            return self._cloud_storage_client
+
+        return self._local_storage_client
+
+    @override
+    def get_additional_cache_key(self, configuration: CrawleeConfiguration) -> Hashable:
+        if self._is_at_home:
+            if isinstance(configuration, ApifyConfiguration):
+                return self._cloud_storage_client.get_additional_cache_key(configuration)
+            raise TypeError('Expecting ApifyConfiguration')
+
+        return self._local_storage_client.get_additional_cache_key(configuration)
+
+    @override
+    async def create_dataset_client(
+        self,
+        *,
+        id: str | None = None,
+        name: str | None = None,
+        alias: str | None = None,
+        configuration: CrawleeConfiguration | None = None,
+    ) -> DatasetClient:
+        return await self._get_suitable_storage_client().create_dataset_client(
+            id=id, name=id, alias=alias, configuration=configuration
+        )
+
+    @override
+    async def create_kvs_client(
+        self,
+        *,
+        id: str | None = None,
+        name: str | None = None,
+        alias: str | None = None,
+        configuration: CrawleeConfiguration | None = None,
+    ) -> KeyValueStoreClient:
+        return await self._get_suitable_storage_client().create_kvs_client(
+            id=id, name=id, alias=alias, configuration=configuration
+        )
+
+    @override
+    async def create_rq_client(
+        self,
+        *,
+        id: str | None = None,
+        name: str | None = None,
+        alias: str | None = None,
+        configuration: CrawleeConfiguration | None = None,
+    ) -> RequestQueueClient:
+        return await self._get_suitable_storage_client().create_rq_client(
+            id=id, name=id, alias=alias, configuration=configuration
+        )
+
+    async def open_dataset(
```

Comment:
These three methods are super internal, it'd be great to mark them as such. The current docblocks make them look like something intended for the users to call